c/parallel/cmake/CParallelHeaderTesting.cmake

```diff
@@ -4,8 +4,12 @@
 # .inl files are not globbed for, because they are not supposed to be used as public
 # entrypoints.
 
-cccl_generate_header_tests(cccl.c.parallel.headers c/parallel/include
+set(target_name cccl.c.parallel.headers)
+
+cccl_generate_header_tests(${target_name} c/parallel/include
   DIALECT 20
+  LANGUAGE C
```

Comment:
Drop DIALECT 20

Reply:
Turns out this is not intended to be an option here. The dialect info is setup by linking with the project target, which propagates dialect info to linked targets.

Removed this and added an error check for unrecognized args to the generate_header_tests function.